### PR TITLE
[Page] Resolve Errors & Warnings

### LINKED ISSUE #333: [Task] Basic skeleton of errors and warnings (Phase 1)
Just to fail early, we want to try to present the user errors and warnings on the page with very basic formatting.

## Current behavior
- no errors and warnings are displayed

## Expected behavior
- errors and warnings are displayed on the page, even if rough and undesigned

### LINKED ISSUE #401: [Page] Multifield Errors

### PR DESCRIPTION
**Pull request by shindigira**
closes #333 
closes #401 

## Note (05/02/24)
The `sbl-filing` branch used is `main` --  latest working commit `0ddf51bd4bd0ab7aefac8cbb02c7797077885620`

branched off [331-filing-routing_phase2-live-data](https://github.com/cfpb/sbl-frontend/pull/406)

## Changes
- feat(Resolve Errors): Single-Field Errors
- feat(Resolve Errors): Multi-Field Errors
- feat(Resolve Errors): Register-Level Errors
- feat(Resolve Errors): Alerts present depending on errors or no errors
- feat(Resolve Errors): Swappable between Step 1 (Syntax Errors) and Step 2 (Logical Errors/Register Errors); navigates to `warnings` on Step 2 (no errors)
- enhancement(Upload): added an `enableLongPolling` boolean to `useGetSubmissionLatest`; default false
- enhancement(Resolve Errors): Implemented 'react-markdown' -- Validation Description
- chore(Resolve Errors): TypeScript handling in `FilingErrors.helpers.ts`
- [chore(Upload): Hide FilingNavButtons if still loading](https://github.com/cfpb/sbl-frontend/pull/423/commits/d380708ee606c249ebd4b552468c4c7a75cb4930)
- feat(Resolve Errors): if a get submission latest fails, there is an Alert that pops up

## Next PRs
- Pagination
- Accessibility pass
- Table styling size adjustments
- Download linking the validation report

## How to Test

1. `yarn install` to install `react-markdown`
2. Select an institution and upload with the csv files provided (Syntax Errors and then Logical/Register Errors)
3. Ensure the right Alert occurs based on the error (or no errors) per segment
4. Verify all proper data is present per field-level section

#### Test CSV's to use
[all-errors-warnings-no-syntax-errors.csv](https://github.com/cfpb/sbl-frontend/files/15182232/all-errors-warnings-no-syntax-errors.csv)
[sblar_syntax_errors_large.csv](https://github.com/cfpb/sbl-frontend/files/15182233/sblar_syntax_errors_large.csv)
[sblar_syntax_errors.csv](https://github.com/cfpb/sbl-frontend/files/15182234/sblar_syntax_errors.csv)

Note: If Syntax Errors are present, then validation for the logical errors/warnings do not occur. Hence, why the test CSV's has either syntax errors or logical errors; not both.

## Screenshots
|Markdown|
|--|
|Screenshot 2024-05-02 at 2 40 05 AM|

| SingleFieldErrors | Success | Register-Level |
|--|--|--|
|Screenshot 2024-05-01 at 4 44 52 PM|Screenshot 2024-05-01 at 4 45 10 PM|Screenshot 2024-05-01 at 4 45 45 PM|

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -26,4 +26,29 @@ https://v1.tailwindcss.com/docs/adding-base-styles#using-css */
...
+    white-space: nowrap;
+}
+
+td:last-child{
+    width: 100%;
+} */
+
+/* td {
+  word-break: break-all;
+} */
```
Comment (meissadia) [inline src/index.css:54]: Assume this gets cleaned up in a later PR? If not, please remove here.

File: src/pages/Filing/FilingApp/FieldEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        // ])}
+        // rows={[
+        //   [
+        //     'Row 1, Column 1',
+        //     'Row 1, Column 123456789TESTBANK123C2  123456789TESTBANK123C2  123456789TESTBANK123C2 123456789TESTBANK123C2 123456789TESTBANK123C2 123456789TESTBANK123C2',
+        //     'Row 1, Column 3 123456789TESTBANK123C2123456789TESTBANK123C2123456789TESTBANK123C2123456789TESTBANK123C2',
+        //   ],
+        //   ['Row 2, Column 1', 'Row 2, Column 2', 'Row 2, Column 3'],
+        //   ['Row 3, Column 1', 'Row 3, Column 2', 'Row 3, Column 3'],
+        // ]}
```
Comment (meissadia) [inline src/pages/Filing/FilingApp/FieldEntry.tsx:65]: Do we still need this test data?

File: src/pages/Filing/FilingApp/FilingErrors/index.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            <InstitutionHeading
+              eyebrow
+              name={institutionName}
+              filingPeriod={year}
+            />
+          </div>
+          <TextIntroduction
+            // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+            heading={`Resolve errors (${isStep2 ? 2 : 1} of 2)`}
+            subheading='First, our system checks that each value in your register meets data type and format requirements. We are unable to accurately detect consequent errors or warnings until each record in your register passes these syntax checks.'
```
Comment (meissadia) [inline src/pages/Filing/FilingApp/FilingErrors/index.tsx:109]: This content is different when `isStep2`.  
---

Next, our system checks your register to confirm that there are no inconsistencies or mistakes in how the information is organized or represented. Your register must pass these logic checks to continue to the next step.

**Review by meissadia — CHANGES_REQUESTED** [review]
Couple cleanup points, but the major find is the Subheading content for Errors step 2.  Looking great though!

File: src/pages/Filing/FilingApp/FilingErrors/index.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            description={
+              <>
+                {isStep2 ? (
+                  <>
+                    Next, our system checks your register to confirm that there
+                    are no inconsistencies or mistakes in how the information is
+                    organized or represented. Your register must pass these
+                    logic checks to continue to the next step.
```
Comment (meissadia) [inline src/pages/Filing/FilingApp/FilingErrors/index.tsx:117]: Sorry should have specified in my initial comment, but this Step 2 content is for the `subheading` not the `description`

Screenshot 2024-05-06 at 1 23 53 PM

**Review by meissadia — CHANGES_REQUESTED** [review]

**Review by meissadia — APPROVED** [review]
👍🏾

**Review by billhimmelsbach — APPROVED** [review]
This is super great! (good eye @meissadia finding the content changes)

Looks like we're missing blue underlines under some links that are in the Figma, could you create a separate ticket or fix that here?
Screenshot 2024-05-06 at 3 55 14 PM
Screenshot 2024-05-06 at 3 55 25 PM

**Comment by billhimmelsbach** [comment]
> We should address this outside of this PR since this styling change could affect all links through the app. 

Sounds good, can we make a ticket for it, then this is good to go @shindigira!

**Pull request merged**